provisioner/terraform/executor.go

```diff
@@ -386,6 +413,61 @@ func (e *executor) showPlan(ctx, killCtx context.Context, planfilePath string) (
 	return p, err
 }
 
+// logDrift must only be called while the lock is held.
+// It will log the output of `terraform show`, which will show which resources have drifted from the known state.
+func (e *executor) logDrift(ctx, killCtx context.Context, planfilePath string, logr logSink) {
+	stdout, stdoutDone := resourceReplaceLogWriter(logr)
+	stderr, stderrDone := logWriter(logr, proto.LogLevel_ERROR)
+	defer func() {
+		_ = stdout.Close()
+		_ = stderr.Close()
+		<-stdoutDone
+		<-stderrDone
+	}()
+
+	err := e.showPlan(ctx, killCtx, stdout, stderr, planfilePath)
+	if err != nil {
+		e.server.logger.Debug(ctx, "failed to log state drift", slog.Error(err))
+	}
+}
+
+// resourceReplaceLogWriter highlights log lines relating to resource replacement by elevating their log level.
+// This will help template admins to visually find problematic resources easier.
+//
+// The WriteCloser must be closed by the caller to end logging, after which the returned channel will be closed to
+// indicate that logging of the written data has finished.  Failure to close the WriteCloser will leak a goroutine.
+func resourceReplaceLogWriter(sink logSink) (io.WriteCloser, <-chan any) {
+	r, w := io.Pipe()
+	done := make(chan any)
```

Comment:
```suggestion
	done := make(chan struct{})
```
Nit: Struct indicates we're not signaling via values and takes up less space 😄